packages/frontend/src/containers/recipe-library/RecipeLibraryContainer.tsx

```diff
@@ -0,0 +1,300 @@
+
+            const newWidget: TUserViewWidget = {
+                id: 990,
+                hash: uuidv4(),
+                name: recipeWidget.name,
+                widget: recipeWidget,
+                settings: recipeWidget.settings.map(({ setting }, id) => ({
+                    widget_setting: {
+                        setting: {
+                            id: Number(id) + 1,
+                            slug: setting.slug,
+                            name: setting.slug,
+                            setting_type: "tags",
+                        },
+                        default_toggle_value: null,
+                        sort_order: 0,
+                    },
+                    tags: [],
+                    toggle_value: false,
+                })),
+                sort_order: selectedView.data.widgets.length,
+            };
+
+            // Create a simple layout with 3 columns and add to the shortest
+            const layoutState: TUserViewWidget[][] = [[], [], []];
+            selectedView.data.widgets.forEach((widget, index) => {
+                layoutState[index % 3].push(widget);
+            });
+
+            const shortestCol = layoutState
+                .map((a) => a.length)
+                .indexOf(Math.min(...layoutState.map((a) => a.length)));
+
+            layoutState[shortestCol].push(newWidget);
+            addWidgetsToCache(recomputeWidgetsPos(layoutState));
+
+            Logger.debug(
+                "RecipeLibraryContainer: Recipe widget added to board"
+            );
+
+            // Mark as added and reset flag
+            setHasAddedRecipeWidget(true);
+            setShouldAddRecipeWidget(false);
+        }
+    }, [
+        shouldAddRecipeWidget,
+        hasAddedRecipeWidget,
+        selectedView,
+        resolvedRecipeWidget,
+        addWidgetsToCache,
+    ]);
+
+    return {
+        triggerAddRecipeWidget: () => setShouldAddRecipeWidget(true),
+    };
+};
+
+export const RecipeLibraryContainer = () => {
+    const { showModal, onClose } = useRecipeLibraryState();
+    const { triggerAddRecipeWidget } = useAddRecipeWidget();
+    const isAuthenticated = useAppSelector(selectIsAuthenticated);
+
+    const {
+        data: recipesData,
+        isLoading: recipesLoading,
+        refetch: refetchRecipes,
+    } = useGetRecipesQuery({});
+    const { data: templatesData, isLoading: templatesLoading } =
+        useGetRecipeTemplatesQuery({});
+    const { data: outputFormatsData } = useGetOutputFormatsQuery({});
+
+    const [createRecipe] = useCreateRecipeMutation();
+    const [updateRecipe] = useUpdateRecipeMutation();
+    const [activateRecipe] = useActivateRecipeMutation();
+    const [deactivateRecipe] = useDeactivateRecipeMutation();
+
+    const onCreateRecipe = (recipe: TRecipeInput) => {
+        if (!isAuthenticated) {
+            toast(globalMessages.callToAction.signUpToBookmark("recipes"));
+            return;
+        }
+
+        const isFirstRecipe =
```

Comment:
I don't think this is sufficient.

If users delete all their recipes, then this will be true again. If it is, then the recipe widget will be duplicated on the AlphaBoard. No?